src/main/java/org/prebid/server/metric/Metrics.java

```diff
@@ -49,12 +45,11 @@
     private final CurrencyRatesMetrics currencyRatesMetrics;
     private final Map<MetricName, SettingsCacheMetrics> settingsCacheMetrics;
 
-    public Metrics(MetricRegistry metricRegistry, CounterType counterType, AccountMetricsVerbosity
-            accountMetricsVerbosity, BidderCatalog bidderCatalog) {
+    public Metrics(MetricRegistry metricRegistry, CounterType counterType,
+                   AccountMetricsVerbosity accountMetricsVerbosity) {
         super(metricRegistry, counterType, MetricName::toString);
 
         this.accountMetricsVerbosity = Objects.requireNonNull(accountMetricsVerbosity);
-        this.bidderCatalog = Objects.requireNonNull(bidderCatalog);
```

Comment:
Where's the bidderCatalog gone?

Reply:
Hi @muuki88 , we have removed Metrics dependency on bidderCatalog for two reasons:

Trying to inject Currency Service to RubiconConfiguration we got cyclic dependency bidderCatalog -> RubiconConfiguration -> CurrencyConversionService -> Metrics -> BidderCatalog. Among few ideas we have, the best was to remove bidderCatalog from Metrics, because it is not required there. BidderCatalog in Metrics used only to validate that bidder has valid name, but actually there is almost no way to pass invalid bidder name to metrics methods, because on stages, where we operate on bidder level, all invalid bidders already dropped. The only one place where were still possible to pass invalid bidder name to metrics was here, in cookieSync handler for rejected by tcf bidders.
https://github.com/prebid/prebid-server-java/pull/1117/files#diff-89b264b2e9293fe0af95ba22db75af31d0dde1beef3e3994f357e8504008742bR386

Reply:
Thanks a lot for the detailed explanation 😍